lib/lyber_core/robot.rb

```diff
@@ -72,21 +72,26 @@ def work(druid)
       #   otherwise, the robot did something like set the step to 'waiting' with a note
 
       # update the workflow status from 'queued' to 'completed' -- errors out if current status is not queued
-      Dor::WorkflowService.update_workflow_status @repo, druid, @workflow_name, @step_name, 'completed', :elapsed => elapsed, :note => Socket.gethostname, :current_status => 'queued'
+      workflow_service.update_workflow_status @repo, druid, @workflow_name, @step_name, 'completed', :elapsed => elapsed, :note => Socket.gethostname, :current_status => 'queued'
       LyberCore::Log.info "Finished #{druid} in #{sprintf("%0.4f",elapsed)}s"
     rescue => e
       begin
         LyberCore::Log.error e.message + "\n" + e.backtrace.join("\n")
-        Dor::WorkflowService.update_workflow_error_status @repo, druid , @workflow_name, @step_name, e.message, :error_text => Socket.gethostname
+        workflow_service.update_workflow_error_status @repo, druid , @workflow_name, @step_name, e.message, :error_text => Socket.gethostname
       rescue => e2
         LyberCore::Log.error "Cannot set #{druid} to status='error'\n" + e2.message + "\n" + e2.backtrace.join("\n")
         raise e2 # send exception to Resque failed queue
       end
     end
 
-  private
+    private
+
+    def workflow_service
+      @workflow_service || Dor::WorkflowService
```

Comment:
I meant `||`. We don't want to hold onto the singleton, and after we have an instantiatable class, I think it should be fairly low-overhead anyway.